File: app/src/main/java/com/simon/harmonichackernews/data/ArxivInfo.java
package com.simon.harmonichackernews.data;

import android.text.TextUtils;

import com.simon.harmonichackernews.utils.ArxivResolver;

public class ArxivInfo {

    public String arxivAbstract;
    public String[] authors;
    public String primaryCategory;
    public String arxivID;

    public String[] secondaryCategories;

    public String publishedDate;

    public String concatNames() {
        return TextUtils.join(", ", authors);
    }

    public String formatDate() {
        return publishedDate.substring(0, 10);
    }

    public String formatSubjects() {
        StringBuilder allSubjects = new StringBuilder(ArxivResolver.resolveFull(primaryCategory));

        for (String secondaryCategory : secondaryCategories) {
            allSubjects.append("; ").append(ArxivResolver.resolveFull(secondaryCategory));
        }
        return allSubjects.toString();
    }

    public String getPDFURL() {
        return "https://arxiv.org/pdf/" + arxivID + ".pdf";
    }

}

File: app/src/main/java/com/simon/harmonichackernews/data/ArxivInfo.kt
package com.simon.harmonichackernews.data

import com.simon.harmonichackernews.utils.ArxivResolver

class ArxivInfo {
    var arxivAbstract: String? = null
    var authors: Array<String?> = emptyArray()
    var primaryCategory: String? = null
    var arxivID: String? = null

    var secondaryCategories: Array<String?> = emptyArray()

    var publishedDate: String? = null

    fun concatNames(): String? = authors.joinToString(", ")

    fun formatDate(): String {
        return publishedDate!!.substring(0, 10)
    }

    fun formatSubjects(): String {
        return buildString {
            append(ArxivResolver.resolveFull(primaryCategory))
            secondaryCategories.forEach { category ->
                append("; ")
                append(ArxivResolver.resolveFull(category))
            }
        }
    }

    val pDFURL: String
        get() = "https://arxiv.org/pdf/$arxivID.pdf"
}
